src/parsing.jl

```diff
@@ -1,267 +1,156 @@
-                        lo = parse(Float64, string(n) * m.captures[5])
-                        hi = parse(Float64, string(n+x) * m.captures[5])
-                        interval = eval(make_interval(T, lo, [hi]))
-                        return interval
-                    end
-
-                    if m.captures[4] == "u" && m.captures[5] == nothing    #strings of the form "3.46?1u"
-                        d = length(m.captures[2])
-                        x = parse(Float64, m.captures[3] * "e-$d")
-                        n = parse(Float64, m.captures[1]*"."*m.captures[2])
-                        lo = n
-                        hi = n+x
-                        interval = eval(make_interval(T, lo, [hi]))
-                        return interval
-                    end
-
-                    if m.captures[4] == "d" && m.captures[5] != nothing  #strings of the form "3.46?1d"
-                        d = length(m.captures[2])
-                        x = parse(Float64, m.captures[3] * "e-$d")
-                        n = parse(Float64, m.captures[1]*"."*m.captures[2])
-                        lo = parse(Float64, string(n-x) * m.captures[5])
-                        hi = parse(Float64, string(n) * m.captures[5])
-                        interval = eval(make_interval(T, lo, [hi]))
-                        return interval
-                    end
-
-                    if m.captures[4] == "d" && m.captures[5] == nothing  #strings of the form "3.46?1d"
-                        d = length(m.captures[2])
-                        x = parse(Float64, m.captures[3] * "e-$d")
-                        n = parse(Float64, m.captures[1]*"."*m.captures[2])
-                        lo = n-x
-                        hi = n
-                        interval = eval(make_interval(T, lo, [hi]))
-                        return interval
-                    end
-
-                    if m.captures[4] == "" && m.captures[5] != nothing    # strings of the form "3.56?1e2"
-                        d = length(m.captures[2])
-                        x = parse(Float64, m.captures[3] * "e-$d")
-                        n = parse(Float64, m.captures[1]*"."*m.captures[2])
-                        lo = parse(Float64, string(n-x)*m.captures[5])
-                        hi = parse(Float64, string(n+x)*m.captures[5])
-                        interval = eval(make_interval(T, lo, [hi]))
-                        return interval
-                    end
-                end
-            end
-
-            m = match(r"(-?\d*\.?\d*)", s)  # match strings of form "1" and "2.4"
-
-            if m != nothing
-                lo = parse(Float64, m.captures[1])
-                hi = lo
-            end
-
-            if m == nothing
-                throw(ArgumentError("Unable to process string $s as interval"))
-            end
-
-            interval = eval(make_interval(T, lo, [hi]))
-            return interval
```

Comment:
Some of these lines treated parsing special strings, such as `"[]"`, `"[ ]"`, `"[entire]"`, `"[ empty ]"`, `"[0x1.3p-1,]"`, `"[,]"`, `"-10??u"`, `"-10??"`, `"[nai]"`, which is not considered anymore. See `tests/ITF1788_tests/ieee1788-constructors.jl` for examples of what is expected.